### PR TITLE
[C++ Worker]Replace so path for mac

### PR DESCRIPTION
**Pull request by qicosmos**
## Why are these changes needed?

A better way to solve problems of c++ worker example running on mac.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests

### CODE REVIEW

File: cpp/example/run.sh [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #!/usr/bin/env bash
 
-ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE:-$0}")" || exit; pwd)"
-
 bazel --nosystem_rc --nohome_rc build //:example
 if [[ "$OSTYPE" == "darwin"* ]]; then
-    DYLD_LIBRARY_PATH="$ROOT_DIR/thirdparty/lib" "${ROOT_DIR}"/bazel-bin/example
+    install_name_tool -change bazel-out/darwin-opt/bin/cpp/libray_api.so ./thirdparty/lib/libray_api.so ./bazel-bin/example.so
+    install_name_tool -change bazel-out/darwin-opt/bin/cpp/libray_api.so ./thirdparty/lib/libray_api.so ./bazel-bin/example
```
Comment (SongGuyang) [inline cpp/example/run.sh:6]: Can we use install_name_tool when we copy `libray_api.so` to `thirdparty` directory? This seems to be a little weird to `install_name_tool ` in users' script.

Comment (qicosmos) [reply cpp/example/run.sh:6]: We need to modify the dependency so path of example and example.so, modify libray_api.so not work.

**Comment by SongGuyang** [comment]
I don't think this way of `install_name_tool ` is the best choice compared to env var `DYLD_LIBRARY_PATH `. @raulchen What do you think?

**Comment by qicosmos** [comment]
> I don't think this way of `install_name_tool ` is the best choice compared to env var `DYLD_LIBRARY_PATH `. @raulchen What do you think?

I think it can solve passing environment variables failed problem of ray wheel package, i will do a test for wheel package on mac.

**Comment by bveeramani** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've switched our code formatter from YAPF to Black (see #21311).

To prevent issues with merging your code, here's what you'll need to do:

1. Install Black
```
pip install -I black==21.12b0
```

2. Format changed files with Black
```
curl -o format-changed.sh https://gist.githubusercontent.com/bveeramani/42ef0e9e387b755a8a735b084af976f2/raw/7631276790765d555c423b8db2b679fd957b984a/format-changed.sh
chmod +x ./format-changed.sh
./format-changed.sh
rm format-changed.sh
```

3. Commit your changes.
```
git add --all
git commit -m "Format Python code with Black"
```

4. Merge master into your branch.
```
git pull upstream master
```

5. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated `format.sh`.

**Pull request closed without merging**